src/Reports/index.ts

```diff
@@ -32,40 +22,47 @@ export class Reports extends Entity {
   }
 
   balanceSheet(filter?: ReportFilter) {
-    return this.generateReport('balanceSheet', filter)
+    return this._generateReport('balanceSheet', filter) as Query<BalanceSheetReport[]>
+  }
+
+  cashflow(filter?: ReportFilter) {
+    return this._generateReport('cashflow', filter) as Query<CashflowReport[]>
   }
 
-  generateReport(type: ReportProcessorType, filter?: ReportFilter) {
+  _generateReport(type: 'balanceSheet' | 'cashflow', filter?: ReportFilter) {
     return this._root._query(
-      [type, ...(filter ? [filter] : [])],
-      () => {
-        const dateFilter = {
-          timestamp: {
-            greaterThan: filter?.from,
-            lessThan: filter?.to,
-          },
-        }
+      [type, filter?.from, filter?.to, filter?.groupBy],
+      () =>
+        defer(() => {
```

Comment:
doesn't need the `defer` here. The callback being called won't immediately fire off requests.

Reply:
hinted at by the fact that the callback inside `defer` isn't an async function